File: solutions/755.reach-a-number/reach-a-number.cpp
class Solution {
public:
    int reachNumber(int target) {
        target = std::abs(target);
        int k = sqrt(target * 2);
        if (sum(k) < target) ++k;
        int d = sum(k) - target;
        if (d % 2 == 0) return k;
        return k + 1 + (k % 2);
    }
private:
    int sum(int k) {
        return k * (k + 1) / 2;
    }
};

File: solutions/755.reach-a-number/reach-a-number.java
class Solution {
    public int reachNumber(int target) {
        target = Math.abs(target);
        int k = 0, sum = 0; 
        while(sum < target) sum += ++k;
        while((sum - target) % 2 != 0) sum += ++k;
        return k;
    }
}